### PR TITLE
Improve github actions triggers

### PR DESCRIPTION
**Pull request by bpar476**
* Run the full test suite every day at midnight
* Run the Snyk security analysis on every push to default branch

### CODE REVIEW

**Comment by bpar476** [comment]
Once merged, I need to double-check that the `secrets.MAIN_BRANCH_REF` works in the github actions configuration.

File: .github/workflows/mvn_test.yml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 name: Tests
 
-on: [push]
-
+on:
+  push:
+  schedule:
+    - cron: "0 0 * * *" # every day at midnight
```
Comment (nghazali) [inline .github/workflows/mvn_test.yml:6]: I think the expression should be  "0 0 0 ? * * " for everyday at midnight.

Comment (bpar476) [reply .github/workflows/mvn_test.yml:6]: My understanding is 0 0 * * * means:
* At minute 0
* At hour 0
* Of every day
* Of every month
* Regardless of the day of the week.

Isn't `0 0 0 ? * * ` too many characters? Also, what's the difference between `?` and `*`. crontab.guru says `0 0 0 ? * *` is invalid https://crontab.guru/#0_0_0_?_*_*

Comment (nghazali) [reply .github/workflows/mvn_test.yml:6]: Ops.. in Quartz corn, expression starts with Seconds so for everyday we need three 0 at the beginning.

**Review by nanux — APPROVED** [review]

**Pull request merged**